### PR TITLE
New page with GitLab-specific instructions

### PR DESCRIPTION
**Pull request by akosma**
This PR provides a new page with instructions about how to connect GitLab to APPUiO Cloud for GitOps scenarios.

Provides a part of the solution for #23.

## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `how-to`, `tutorial`, `reference`, `explanation`, `cicd`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to/connect-gitlab.adoc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+----
+
+=== On GitLab (via Certificates)
+
+The final step consists of the following:
+
+. In your GitLab project, select the menu:Infrastructure[Kubernetes clusters] menu item, and scroll to the bottom to select the btn:[Connect existing cluster] button.
+** You can also select the menu:Actions[Connect with a certificate] menu entry, avilable at the right side of the btn:[Actions] button.
+. Fill the form with the following information:
+** Kubernetes cluster name: "APPUiO Cloud."
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:48]: I would add the zone name here as well.

**Review by tobru — APPROVED** [review]
LGTM - Let's have another review (I pinged some VSHNeers)

File: docs/modules/ROOT/pages/how-to/connect-gitlab.adoc (hunk 1 above)
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc]: I just added it.

**Review by ccremer — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-to/connect-gitlab.adoc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+** You can also select the menu:Actions[Connect with a certificate] menu entry, available at the right side of the btn:[Actions] button.
+. Fill the form with the following information:
+** Kubernetes cluster name: "APPUiO Cloud [ZONE NAME]."
+** API URL: Use the xref:references/zones.adoc[URL corresponding to your chosen zone].
+** CA Certificate: leave empty.
+** Service Token: paste the value copied in the `oc get secret` command above.
+** Check the "RBAC-enabled cluster" option.
+** Uncheck "GitLab-managed cluster" and "Namespace per environment."
+** Leave the "Project namespace prefix (optional, unique)" empty.
+. Click the btn[Add Kubernetes cluster]
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:55]:
```suggestion
. Click the button btn:[Add Kubernetes cluster]
```

File: docs/modules/ROOT/pages/how-to/connect-gitlab.adoc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    spec:
+      imagePullSecrets:
+        - name: gitlab-pull-secret # <1>
+      containers:
+        - name: xxxx
+          image: xxxxxxxxxx
+          # …
+----
+
+<1> Name of the secret created in the previous step.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:121]: Maybe only use the minimal path to spec?
Besides, we should probably not encourage the use of `DeploymentConfig` anymore?
```suggestion
apiVersion: apps/v1
kind: Deployment
metadata:
  name: deployment-name
spec:
  template:
    spec:
      imagePullSecrets:
        - name: gitlab-pull-secret # <1>
      ...
----
<1> Name of the secret created in the previous step.
```

**Review by corvus-ch — APPROVED** [review]
Nothing left for me. @ccremer did cover it all. As soon as his points are resolved, this is fine for mee too.

File: docs/modules/ROOT/pages/how-to/connect-gitlab.adoc [hunk 4]
```diff
@@ -0,0 +1,121 @@
+= Connect GitLab to APPUiO Cloud
+
+This page explains how to connect your GitLab project to APPUiO Cloud for your GitOps operations.
```
[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: This breaks the current version of the search engine, sadly. The title of the search engine result says literally "Connect GitLab to {product}" in that case. I don't agree this is a good idea, to be honest. Why is this a variable? How often are we going to change this value? I don't get it.

[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: "makes typing a bit easier" => I doubt it. 😆

[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: I think both sections in this document are GitOps specific, so the description applies correctly.

[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: Search engine results:

Selection_001

[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: I just opened #61 to address this point in detail. I'm closing this conversation for now.

[the remark this answers is not in the document]
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/connect-gitlab.adoc:3]: `vshn/antora-indexer-cli:1.12` solves this issue.

**Comment by akosma** [comment]
Ready to merge, @ccremer could you please check again? Thanks.

**Review by ccremer — APPROVED** [review]

**Pull request merged**